wallet/src/de/schildbach/wallet/ui/dashpay/CreateIdentityService.kt

```diff
@@ -317,19 +305,35 @@ class CreateIdentityService : LifecycleService() {
         platformRepo.resetIdentityCreationStateError(blockchainIdentityData)
 
         val wallet = walletApplication.wallet!!
-        val password = securityGuard.retrievePassword()
-
 
-        val encryptionKey = deriveKey(backgroundHandler, wallet, password)
+        val encryptionKey = platformRepo.getWalletEncryptionKey() ?: throw IllegalStateException("cannot obtain wallet encryption key")
 
         if (blockchainIdentityData.creationState <= CreationState.UPGRADING_WALLET) {
             platformRepo.updateIdentityCreationState(blockchainIdentityData, CreationState.UPGRADING_WALLET)
-            val seed = decryptSeed(backgroundHandler, wallet, encryptionKey)
-            platformRepo.addWalletAuthenticationKeysAsync(seed, encryptionKey)
+            val seed = platformRepo.getWalletSeed() ?: throw IllegalStateException("cannot obtain wallet seed")
+            platformRepo.addWalletKeyChainsAsync(seed, encryptionKey)
         }
 
         val blockchainIdentity = platformRepo.initBlockchainIdentity(blockchainIdentityData, wallet)
+        // look for the credit funding tx in case there was an error in the next step previously
+        for (tx in wallet.creditFundingTransactions) {
+            tx as CreditFundingTransaction
```

Comment:
This is a weird expression. Perhaps it should be a check `if (tx is CreditFundingTransaction)`